feat: Adding ClipComponent (#1769)
flame-engine/flame: doc/flame/components.md

[custom_painter_component](https://github.com/flame-engine/flame/blob/main/examples/lib/stories/widgets/custom_painter_example.dart)
for details on how to use it.

## ClipComponent

A `ClipComponent` is a component that will clip the canvas to its size and shape. This means that

> if the any child of a `ClipComponent` is beign rendered outside of its bondaries, the part that is

Reviewer: ```suggestion
if the component itself or any child of the `ClipComponent` renders outside of the `ClipComponent`'s boundaries, the part that is
```

> if the component itself or any child of the `ClipComponent` renders outside of the `ClipComponent`'s boundaries, the part that is

Reviewer: This line became too long ofter my suggestions. :sweat_smile:

not inside the area will not be shown.

> There are three types of shapes that a `ClipComponent` can take form:

Reviewer: ```suggestion
There are three types of shapes that a `ClipComponent` can use to define the area that shoud be clipped:
```

 - `ClipComponent.rect`: Clips the area in the form a rectangle based on its size.
 - `ClipComponent.circle`: Clips the area in the form of a circle based on its size.
 - `ClipComponent.poylon`:  Clips the area in the form of a polygon based on the points received

> on the constructor.

Reviewer: ```suggestion
in the constructor.
```

>  - `ClipComponent.poylon`:  Clips the area in the form of a polygon based on the points received

Reviewer 2: ```suggestion
 - `ClipComponent.polygon`:  Clips the area in the form of a polygon based on the points received
```

[custom_painter_component](https://github.com/flame-engine/flame/blob/main/examples/lib/stories/widgets/custom_painter_example.dart)
for details on how to use it.

## ClipComponent

A `ClipComponent` is a component that will clip the canvas to its size and shape. This means that
if the component itself or any child of the `ClipComponent` renders outside of the
`ClipComponent`'s boundaries, the part that is not inside the area will not be shown.

> There are three types of shapes that a `ClipComponent` can use to define the area that shoud be clipped:

Reviewer: Too long line

[custom_painter_component](https://github.com/flame-engine/flame/blob/main/examples/lib/stories/widgets/custom_painter_example.dart)
for details on how to use it.

## ClipComponent

A `ClipComponent` is a component that will clip the canvas to its size and shape. This means that
if the component itself or any child of the `ClipComponent` renders outside of the
`ClipComponent`'s boundaries, the part that is not inside the area will not be shown.

There are three types of shapes that a `ClipComponent` can use to define the area that shoud be
clipped:

>  - `ClipComponent.rect`: Clips the area in the form a rectangle based on its size.

Reviewer 3: I think this requires to be updated
